### PR TITLE
ci: dynamic e2e timeout

### PR DESCRIPTION
**Pull request by bergeron**
## **Description**

I have a PR that changes several long running e2e test suites.  And due to https://github.com/MetaMask/metamask-extension/pull/24556 re-running changed test files extra times, the runs were exceeding the 20 minute timeout.

This PR makes the timeout `15 minutes, plus 3 minutes for every changed file, up to a maximum of 30 minutes`.

(A lot of the changed lines are just converting from `await fs.readFile` to `fs.readFileSync` and removing `async/await`. This was necessary to do because `.circleci/scripts/test-run-e2e-timeout-minutes.ts` needs to `console.log` the result before it ends, and you can't put `await` in the main thread of a Node program.)

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27146?quickstart=1)

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by bergeron** [comment]
Thanks for the improvements Howard

**Review by HowardBraham — APPROVED** [review]

**Review by seaona — APPROVED** [review]

File: .circleci/scripts/test-run-e2e-timeout-minutes.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { filterE2eChangedFiles } from '../../test/e2e/changedFilesUtil';
+
+const changedOrNewTests = filterE2eChangedFiles();
+
+//15 minutes, plus 3 minutes for every changed file, up to a maximum of 30 minutes
+const extraTime = Math.min(15 + changedOrNewTests.length * 3, 30);
+
+console.log(extraTime);
```
Comment (davidmurdoch) [inline .circleci/scripts/test-run-e2e-timeout-minutes.ts:8]: _Ackchyually_ this isn't "extra time"; it's a timeout time _plus_ extra time. :-)

Not a show stopper for the PR, by any means. This PR is two changes: reduce the default timeout time from 20 minutes to 15 minutes, and add 3 minutes of extra time for each changed e2e file.


Reducing the time from 20 to 15 is what I'm curious out. How did you arrive at 15?

**Review by davidmurdoch — APPROVED** [review]
approved with a question

File: .circleci/scripts/test-run-e2e-timeout-minutes.ts (hunk 1 above)
Comment (bergeron) [reply .circleci/scripts/test-run-e2e-timeout-minutes.ts:8]: @HowardBraham did the implementation and might know why

Comment (davidmurdoch) [reply .circleci/scripts/test-run-e2e-timeout-minutes.ts:8]: cool. I'm 👍  on giving it a go with 15 minutes

**Review by vthomas13 — APPROVED** [review]

**Review by desi — APPROVED** [review]
Approving as a Library Admin only. Has been reviewed and discussed by the team so feel ok doing that.

**Pull request merged**